mars/services/scheduling/supervisor/assigner.py

```diff
@@ -164,7 +191,18 @@ async def assign_subtasks(self, subtasks: List[Subtask]):
                         max_size = size
                     elif size == max_size:
                         bands.append(band)
-            assigns.append(random.choice(bands))
+            band = random.choice(bands)
+            if band in exclude_bands and exclude_bands_force:
+                raise NoAvailableBand(
+                    f"No bands available for subtask {subtask.subtask_id} after "
+                    f"excluded {exclude_bands}"
+                )
+            if subtask.bands_specified and band not in subtask.expect_bands:
+                raise Exception(
```

Comment:
Why use Exception instead of NoAvailableBand?

Reply:
fixed